File: infra-market-server/src/main/kotlin/io/infra/market/util/AuthHolder.kt
package io.infra.market.util

/**
 * 认证相关的ThreadLocal工具类
 * @author liuqinglin
 * Date: 2025/8/30
 */
object AuthHolder {
    
    private val uidHolder = ThreadLocal<Long>()
    
    /**
     * 获取当前用户ID
     */
    fun getUid(): Long? = uidHolder.get()
    
    /**
     * 设置当前用户ID
     */
    fun setUid(userId: Long) = uidHolder.set(userId)

    /**
     * 清理当前用户ID
     */
    fun clearUid() = uidHolder.remove()
}

File: infra-market-server-go/internal/util/auth_holder.go
package util

// AuthHolder 认证相关的Context工具
type AuthHolder struct{}

var uidKey = "uid"

// GetUID 从Context获取当前用户ID
func GetUID(ctx any) (uint64, bool) {
	// 这里需要根据实际的Context类型来实现
	// 在Gin中，应该从gin.Context获取
	return 0, false
}

// SetUID 设置当前用户ID到Context
func SetUID(ctx any, uid uint64) {
	// 在Gin中，应该设置到gin.Context
}

// ClearUID 清理当前用户ID
func ClearUID(ctx any) {
	// 在Gin中，不需要特别清理
}
